lib/runtime/addon.js

```diff
@@ -341,4 +340,34 @@ export default class Addon {
     return configModule && (configModule.default || configModule);
   }
 
+  get blueprintNames() {
+    let names = fs.readdirSync(path.join(this.dir, 'blueprints'));
+    return names;
+  }
+
+  get blueprints() {
+    return this.blueprintNames.map((name) => {
+      let blueprintPath = path.join(this.dir, 'blueprints', name);
+      let Print = require(blueprintPath).default;
+      return new Print({ name, addon: this, description: Print.description });
```

Comment:
Perhaps we should build this array eagerly when the Addon is instantiated? Or at least memoize this getter? Seems like it might be a source of rather surprising / difficult to diagnose bugs in the future: when you use `addon.blueprints` more than once, you get a different set of objects representing the same blueprints each time.